frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/projects/ProjectPortlet.java

```diff
@@ -262,6 +269,76 @@ private void importBom(ResourceRequest request, ResourceResponse response) {
         }
     }
 
+    private void createClearingRequest(ResourceRequest request, ResourceResponse response) throws PortletException {
+        User user = UserCacheHolder.getUserFromRequest(request);
+        ClearingRequest clearingRequest = null;
+        AddDocumentRequestSummary requestSummary = null;
+        try {
+            JsonNode crNode = OBJECT_MAPPER.readValue(request.getParameter(CLEARING_REQUEST), JsonNode.class);
+            clearingRequest = OBJECT_MAPPER.convertValue(crNode, ClearingRequest.class);
+            clearingRequest.setRequestingUser(user.getEmail());
+            clearingRequest.setClearingState(ClearingRequestState.NEW);
+            LiferayPortletURL projectUrl = createDetailLinkTemplate(request);
```

Comment:
Can this URL formation be done in jsp.

Reply:
I don't think forming the URL in JSP (using JavaScript) and sending it to back-end will have any significance over the current logic.
Hence leaving the logic as is.